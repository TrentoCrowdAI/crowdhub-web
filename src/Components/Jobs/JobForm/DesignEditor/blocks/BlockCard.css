.card-header a {
    display: flex;
    cursor: pointer;
}

.card-header a i.fa-caret-down {
    margin: auto 0 auto auto;
    transition-duration: 0.5s;
}

i.fa-caret-down.expanded {
    transform: rotate(180deg);
}
