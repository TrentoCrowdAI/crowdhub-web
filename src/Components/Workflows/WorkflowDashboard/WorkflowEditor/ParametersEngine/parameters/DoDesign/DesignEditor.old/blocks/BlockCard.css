.card-header > div {
    display: flex;
    cursor: pointer;
}

.card-header i.fa-caret-down {
    margin: auto 0 auto auto;
    transition-duration: 0.5s;
}

.card-header i.fa-exclamation-triangle {
    margin-right: .5em;
    margin-top: .25em;
}


i.fa-caret-down.expanded {
    transform: rotate(180deg);
}

