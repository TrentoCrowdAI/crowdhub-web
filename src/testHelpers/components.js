import LoadingContainer from "../Components/common/LoadingContainer";

export const expectComponent = (wrapper, Component) => expect(
  wrapper.find(Component).length
).toBe(1);


export const expectIsLoading = (wrapper) => expect(
  wrapper.find(LoadingContainer).first().prop('loading')
).toBe(true);
