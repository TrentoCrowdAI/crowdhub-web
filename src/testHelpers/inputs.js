function expectElementToHaveAttribute (element, attributeName, attributeValue) {
  expect(element.prop(attributeName)).toEqual(attributeValue);
}

export function expectElementToHaveValue (element, value) {
  expectElementToHaveAttribute(element, 'value', value);
}

export function expectInputToHaveValue (wrapper, field, value) {
  expectElementToHaveValue(wrapper.find(`input[name="${field}"]`), value);
}

export function expectTextAreaToHaveValue (wrapper, field, value) {
  expectElementToHaveValue(wrapper.find(`textarea[name="${field}"]`), value);
}

export function expectCheckboxToHaveValue (wrapper, field, checked) {
  expectElementToHaveAttribute(wrapper.find(`input[name="${field}"]`), 'checked', checked);
}

export function expectSelectToHaveValue (wrapper, field, value) {
  expectElementToHaveValue(wrapper.find(`select[name="${field}"]`), value);
}

export const simulateBlurOnAnInput = wrapper => wrapper.find('input').first().simulate('blur');
