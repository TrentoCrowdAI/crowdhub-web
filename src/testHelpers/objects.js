// TODO: Generalize
export const objectClonerWithoutField = object => field => {
  const clone = {...object};
  delete clone[field];
  return clone;
};
