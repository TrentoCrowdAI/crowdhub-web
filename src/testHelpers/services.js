export const createMockForServiceMethod = (service, method) => toReturn => {
  const mock = jest.fn(() => new Promise(resolve => resolve(toReturn)));
  service[method] = mock;
  return mock;
};
