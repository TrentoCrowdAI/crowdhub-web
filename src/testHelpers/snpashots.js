import renderer from "react-test-renderer";

export const expectToMatchSnapshot = component => expect(
  renderer.create(component).toJSON()
).toMatchSnapshot();
