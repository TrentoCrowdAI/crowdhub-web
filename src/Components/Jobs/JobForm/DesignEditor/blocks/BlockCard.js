import React, {Component} from "react";
import {Card, Collapse} from "react-bootstrap";
import './BlockCard.css';


export default class BlockCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      expanded: props.expanded
    };
  }

  toggleExpansion = () => this.setState(
    (previousState) => {
      return {expanded: !previousState.expanded}
    },
    () => this.props.onToggleExpansion(this.state.expanded)
  );

  render() {
    return (
      <Card className="block-card" data-block-type={this.props.type} data-block-id={this.props.id}>
        <Card.Header>
          {
            this.props.expandable &&
            <a onClick={this.toggleExpansion}>
              {
                /* Hacky: show the warning icon only if card is expandable */
                this.props.valid === false &&
                <i className="fas fa-exclamation-triangle"/>
              }

              {this.props.title}

              <i className={`fas fa-caret-down ${this.state.expanded ? 'expanded' : 'collapsed'}`}/>
            </a>
          }
          {!this.props.expandable && this.props.title}

        </Card.Header>
        <Collapse in={this.state.expanded}>
          <div>
            <Card.Body>
              {this.props.children}
            </Card.Body>
          </div>
        </Collapse>
      </Card>
    );
  }
}
