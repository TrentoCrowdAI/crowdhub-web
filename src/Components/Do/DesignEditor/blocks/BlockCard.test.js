import React from 'react';
import BlockCard from "./BlockCard";
import {expectToMatchSnapshot} from "../../../../testHelpers/snpashots";

const MockedContent = () => <p>mocked card content</p>;

const makeCardBlock = (card) => (
  <BlockCard {...card} expandable="false" title="Mocked Block">
    <MockedContent/>
  </BlockCard>
);

const expectCardBlockToMatchSnapshot = (card) => expectToMatchSnapshot(makeCardBlock(card));

it('renders a collapsed not-expandable BlockCard', () =>
  expectCardBlockToMatchSnapshot({
    type: 'mock',
    id: '1',
    expanded: false
  }));


it('renders a collapsed expandable BlockCard', () =>
  expectCardBlockToMatchSnapshot({
    type: 'mock',
    id: '1',
    expanded: false
  }));

it('renders a collapsed expandable BlockCard with not valid data', () =>
  expectCardBlockToMatchSnapshot({
    type: 'mock',
    id: '1',
    expanded: false,
    valid: false
  }));

it('renders a expanded BlockCard', () =>
  expectCardBlockToMatchSnapshot({
    type: 'mock',
    id: '1',
    expanded: true
  }));

