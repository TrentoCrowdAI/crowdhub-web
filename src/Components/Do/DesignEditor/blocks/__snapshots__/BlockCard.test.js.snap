// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a collapsed expandable BlockCard 1`] = `
<div
  className="block-card card"
  data-block-id="1"
  data-block-type="mock"
>
  <div
    className="card-header"
  >
    <div
      onClick={[Function]}
    >
      <div>
        Mocked Block
      </div>
      <i
        className="fas fa-caret-down collapsed"
      />
    </div>
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      <p>
        mocked card content
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a collapsed expandable BlockCard with not valid data 1`] = `
<div
  className="block-card card"
  data-block-id="1"
  data-block-type="mock"
>
  <div
    className="card-header"
  >
    <div
      onClick={[Function]}
    >
      <div>
        <i
          className="fas fa-exclamation-triangle"
        />
        Mocked Block
      </div>
      <i
        className="fas fa-caret-down collapsed"
      />
    </div>
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      <p>
        mocked card content
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a collapsed not-expandable BlockCard 1`] = `
<div
  className="block-card card"
  data-block-id="1"
  data-block-type="mock"
>
  <div
    className="card-header"
  >
    <div
      onClick={[Function]}
    >
      <div>
        Mocked Block
      </div>
      <i
        className="fas fa-caret-down collapsed"
      />
    </div>
  </div>
  <div
    aria-expanded={null}
    className="collapse"
  >
    <div
      className="card-body"
    >
      <p>
        mocked card content
      </p>
    </div>
  </div>
</div>
`;

exports[`renders a expanded BlockCard 1`] = `
<div
  className="block-card card"
  data-block-id="1"
  data-block-type="mock"
>
  <div
    className="card-header"
  >
    <div
      onClick={[Function]}
    >
      <div>
        Mocked Block
      </div>
      <i
        className="fas fa-caret-down expanded"
      />
    </div>
  </div>
  <div
    aria-expanded={null}
    className="collapse show"
  >
    <div
      className="card-body"
    >
      <p>
        mocked card content
      </p>
    </div>
  </div>
</div>
`;
